app/Services/ArticleService.php

```diff
@@ -37,6 +37,10 @@ public function getCategorizedArticlesByMagazine($magazineId)
         // For now, simply sorting categories alphabetically.
         ksort($categories);
 
+        foreach ($categories as $index => $category) {
+            $categories[$index] = collect($category)->sortByDesc('created_at');
+        }
```

Comment:
I don't see the benefit in creating one more class just for sorting, if that is what you're trying to say. Maybe we can discuss this over slack, as fairly new to all these principles, I wish that you could guide me.